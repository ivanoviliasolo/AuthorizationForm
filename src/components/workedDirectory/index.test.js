import Enzyme, {shallow} from 'enzyme';
import React from 'react';
import {WorkedDirectory} from './index';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });


describe('Тестируем компонент WorkedDirectory ',()=>{
    
    it('Тест на текст возвращаемого значения компонента, должен совпадать ожиданиям', ()=>{
        const nextProps={
            registratedUsers:[{login:'Admin'}],
            dispatchAction:()=>{},
    }
        const WorkedDirectoryTest = shallow(<WorkedDirectory  {...nextProps}/>)
        expect(WorkedDirectoryTest.find('.wrapp-main').text())
        .toEqual("Добрый день -"+nextProps.registratedUsers[0].login+"Выйти из учетной записи "+nextProps.registratedUsers[0].login)
    })

    it('Тестирование с помощью снимков на изменения в структуре компонента', ()=>{
        const nextProps={
            registratedUsers:[{login:'Admin'}],
            dispatchAction:()=>{},
    }
        const Component = renderer.create(<WorkedDirectory {...nextProps}/>).toJSON();

        expect(Component).toMatchSnapshot();
    });
})