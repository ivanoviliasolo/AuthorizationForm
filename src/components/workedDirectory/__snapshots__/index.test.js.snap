// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестируем компонент WorkedDirectory  Тестирование с помощью снимков на изменения в структуре компонента 1`] = `
<div
  className="wrapp-main"
>
  Добрый день -
  Admin
  <button
    className="unlogin-button"
    onClick={[Function]}
  >
    Выйти из учетной записи
     
    Admin
  </button>
</div>
`;
